self_hosted/bug_workaround.jou

```diff
@@ -0,0 +1,15 @@
+# TODO: these shouldn't be needed
+
+if WINDOWS:
+    def is_windows() -> bool:
+        return True
+else:
+    def is_windows() -> bool:
+        return False
```

Comment:
I think I'll fix in this PR. But there is another problem, if you do `if WINDOWS: call_some_windows_only_func()` inside a function/method that will not work, and I will probably create issue about it and fix later

Reply:
Why not do:
```python
def is_windows() -> bool:
    if WINDOWS:
        return True
    return False
```
?

Reply:
That doesn't work yet. The code for handling `if` statements outside functions is not the same code that handles `if` statements inside functions. Only the code for `if` statements outside functions knows about `WINDOWS` and `MACOS`.

As I wrote above (though not very clearly, it was around 4AM), I will fix this before I merge this PR.

Reply:
This now works. In fact, I deleted `is_windows()` because it was unnecessary.